proposals/0018/README.md

```diff
@@ -0,0 +1,191 @@
+---
+id: 0018
+title: Add Instance Object Definition
+status: ideation
+authors: Manny <mmendez@equinix.com>
+---
+
+## Summary
+
+The current hardware data model does not support any OS installation details.
+Installing an OS is one of the main focus areas of the Tinkerbell project at the moment.
+This exists in the legacy Packet Platform under "instance" in the hardware object and a similar concept is in [metal3].
+
+This would be used by Boots and Hegel (at least) to serve their needs.
+
+## Goals and not Goals
+
+We have a goal of keeping the Hardware data relatively static.
+We need a locaion to place instance/provision/installation-info that changes from provision to provision.
+It is not a goal to stick this info in `elastic_data`.
+
+## Content
+
+This proposal is about adding a storage concept for data that is not as permanent as Hardware, and that can persist across multiple Workloads.
+This would mimic the EquinixMetal life-cycle of a machine, provisioned and thus occupied until the customer deletes and the machine is deprovisioned.
```

Comment:
I think having more information about the expected lifecycle of the proposed "instance" type would help.

A few questions that I have:
- Is the Instance expected to be user-created, created through a workflow, or other automated means?
- How is the Instance linked to underlying hardware?
- How is appropriate hardware chosen for an instance?